purchase_work_acceptance/models/purchase.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.addons import decimal_precision as dp
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    wa_count = fields.Integer(
+        compute='_compute_wa',
+        string='WA count',
+        default=0,
+        store=True
+    )
+    wa_ids = fields.Many2many(
+        comodel_name='work.acceptance',
+        compute='_compute_wa',
+        string='Work Acceptances',
+        copy=False,
+        store=True,
+    )
+    wa_lines = fields.One2many(
+        comodel_name='work.acceptance.line',
+        inverse_name='purchase_line_id',
+        string="WA Lines",
+        readonly=True,
+        copy=False
+    )
+
+    @api.depends('order_line.wa_lines.wa_id')
+    def _compute_wa(self):
+        for order in self:
+            wa = self.env['work.acceptance']
+            for line in order.order_line:
+                wa |= line.wa_lines.mapped('wa_id')
+            order.wa_ids = wa
+            order.wa_count = len(wa)
+
+    @api.multi
+    def action_view_wa(self):
+        act = self.env.ref('purchase_work_acceptance.action_work_acceptance')
+        result = act.read()[0]
+        create_wa = self.env.context.get('create_wa', False)
+        result['context'] = {
+            'default_purchase_id': self.id,
+            'default_partner_id': self.partner_id.id,
+            'default_partner_ref': self.partner_ref,
+            'default_currency_id': self.currency_id.id,
+            'default_company_id': self.company_id.id,
```

Comment:
I saw in field company_id, you already have default value for it. So, this wouldn't need.
```suggestion
            'default_company_id': self.company_id.id,
```
Note: and also, why we have both `default_company_id` and `company_id` here?